homework/index.js

```diff
@@ -30,18 +31,102 @@
+      while (mainContainerForRepo.firstChild) {
+        mainContainerForRepo.removeChild(mainContainerForRepo.firstChild);
+      }
+      createDescription(selectedRepository, root);
+      createContributors(selectedRepository, root);
+    });
+  }
+  const mainContainerForRepo = document.querySelector('.container');
+
+  function createDescription(selectedRepository) {
+    const descriptionContainer = createAndAppend('div', mainContainerForRepo, {
+      class: 'left-div',
+    });
+    const table = createAndAppend('table', descriptionContainer, { class: 'table' });
+    const tbody = createAndAppend('tbody', table);
+    const details = [
+      { title: 'Repository', value: selectedRepository.name },
+      { title: 'Description', value: selectedRepository.description },
+      { title: 'Forks', value: selectedRepository.forks },
+      { title: 'Updated', value: selectedRepository.updated_at },
+    ];
+    details.forEach(detail => {
+      const tr = createAndAppend('tr', tbody);
+      createAndAppend('td', tr, {
+        text: detail.title,
+        class: `label `,
+      });
+      createAndAppend('td', tr, { text: detail.value, class: 'repository-data' });
+    });
+  }
+
+  function createContributors(selectedRepository, root) {
+    const contributorsContainer = createAndAppend('div', mainContainerForRepo, {
+      class: 'right-div',
+    });
+    createAndAppend('h3', contributorsContainer, {
+      text: 'Contributors',
+      class: 'contributors-header',
+    });
+    const ul = createAndAppend('ul', contributorsContainer, { class: 'contributor-list' });
+    fetchJSON(selectedRepository.contributors_url, (err, ShowContributors) => {
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
+        ShowContributors.forEach(contributor => {
+          const li = createAndAppend('li', ul, {
+            class: 'contributor-item',
+          });
+          const alink = createAndAppend('a', li);
+          const dataDiv = createAndAppend('div', alink, { class: 'contributor' });
+
+          createAndAppend('img', dataDiv, {
+            class: 'contributor-item',
+            src: contributor.avatar_url,
+            height: 52,
+          });
+          const cotributorData = createAndAppend('div', dataDiv, {
+            class: 'contributor-data',
+          });
+          createAndAppend('div', cotributorData, {
+            class: 'contributor-login',
+            text: contributor.login,
+          });
+          createAndAppend('div', cotributorData, {
+            class: 'contributor-badge',
+            text: contributor.contributions,
+          });
+        });
+      }
+    });
+  }
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
```

Comment:
It is best to exit early from the callback function if you detect an error here. This saves you from having to track down where the `else` clause should end and also saves you some indentation:

```js
if (err) {
  createAndAppend('div', root, { text: err.message, class: 'alert-error' });
  return;
}
creatHeader(data, root);
...
```

You can do the same thing in your `createContributors()` function.